trappy/ftrace.py

```diff
@@ -43,6 +44,37 @@ def _plot_freq_hists(allfreqs, what, axis, title):
         trappy.plot_utils.plot_hist(allfreqs[actor], ax, this_title, "KHz", 20,
                              "Frequency", xlim, "default")
 
+def _trace_cache_path(trace_file):
+    cache_dir  = '.' +  os.path.basename(trace_file) + '.cache'
+    cache_path = os.path.dirname(trace_file) +  '/' + cache_dir
+    return cache_path
+
+def _check_trace_cache(trace_file):
+    cache_path = _trace_cache_path(trace_file)
+    md5file = cache_path + '/md5sum'
+
+    if not os.path.exists(cache_path) or not os.path.exists(md5file):
+        return False
+
+    md5sum = open(md5file, 'r').read()
+    # check md5sum
+    if hashlib.md5(open(trace_file, 'rb').read()).hexdigest() != md5sum:
+        # Cache is invalid, delete it
+        shutil.rmtree(cache_path)
+        return False
+    return True
+
+def _create_trace_cache(trace_file):
+    cache_path = _trace_cache_path(trace_file)
+    md5file = cache_path + '/md5sum'
+
+    if os.path.exists(cache_path):
+        shutil.rmtree(cache_path)
+    os.mkdir(cache_path)
+
+    md5sum = hashlib.md5(open(trace_file, 'rb').read()).hexdigest()
+    open(md5file, 'w').write(md5sum)
+
```

Comment:
I felt these are just utility methods, that didn't as such belong to these classes so I'm just as happy keeping them in the module than within classes